assets/js/util.js

```diff
@@ -293,15 +298,74 @@ function callApy(options, endpoint) {
     var requestUrl = window.location.protocol + window.location.hostname +
         window.location.pathname + '?' + $.param(requestOptions.data);
 
+    callApyStartTime = Date.now();
     if(requestUrl.length > THRESHOLD_REQUEST_URL_LENGTH) {
         requestOptions.type = 'POST';
         return $.ajax(requestOptions);
     }
     return $.jsonp(requestOptions);
 }
 
+function callApyDuration() {
+    callApyEndTime = Date.now();
+    checkServiceLoadTimes(callApyEndTime - callApyStartTime);
+}
+
+function checkServiceLoadTimes(requestDuration) {
+    var individualDurationThreshold = 2000;
+    var cumulativeDurationThreshold = 1500;
+    var demoThreshold = 10000; // for demo purposes only
+    if(store.able()) {
+        if(sessionStorage.requestsMade) {
+            sessionStorage.requestsMade = Number(sessionStorage.requestsMade) + 1;
+        }
+        else {
+            sessionStorage.requestsMade = 1;
+        }
+
+        if(sessionStorage.cumulativeRequestsTime) {
+            sessionStorage.cumulativeRequestsTime = Number(sessionStorage.cumulativeRequestsTime) + requestDuration;
+        }
+        else {
+            sessionStorage.cumulativeRequestsTime = requestDuration;
+        }
+
+        var averageRequestsDuration = (sessionStorage.cumulativeRequestsTime) / (sessionStorage.requestsMade);
+
+        if(requestDuration > individualDurationThreshold || averageRequestsDuration > cumulativeDurationThreshold) {
```

Comment:
> Um, maybe because the network load may not necessarily stay the same over time. So if someone makes a request after certain time, assuming the load may have changed, we could start the process afresh. This is what I thought though! Makes sense?

I don't think that's necessary. A moving average serves that purpose since old values get discarded.

> Um, could you elaborate this a bit more?

Which part? This feature is taking a bit longer than ideal so feel free to just implement the second criteria (how many). Don't forget to also work on the other features (e.g. website translation) concurrently if you are blocked on this one.